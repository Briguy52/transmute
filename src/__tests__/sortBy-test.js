import { List, Map } from 'immutable';
import sortBy from '../sortBy';

describe('transmute/sortBy', () => {
  const modSort = sortBy(n => n % 2);

  it('sorts an Array', () => {
    expect(modSort([1, 2, 3, 10])).toEqual([2, 10, 1, 3]);
  });

  it('sorts a List', () => {
    expect(modSort(List.of(1, 2, 3, 10))).toMatchSnapshot();
  });

  it('sorts a Map into an OrderedMap', () => {
    expect(modSort(Map({ one: 1, two: 2, three: 3 }))).toMatchSnapshot();
  });

  it('sorts an Object', () => {
    expect(modSort({ one: 1, two: 2, three: 3, ten: 10 })).toEqual({
      two: 2,
      ten: 10,
      one: 1,
      three: 3,
    });
  });
});
