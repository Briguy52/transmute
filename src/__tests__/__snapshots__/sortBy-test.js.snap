// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transmute/sortBy sorts a List 1`] = `
Immutable.List [
  2,
  10,
  1,
  3,
]
`;

exports[`transmute/sortBy sorts a Map into an OrderedMap 1`] = `
Immutable.OrderedMap {
  "two": 2,
  "one": 1,
  "three": 3,
}
`;
